AlphaWallet/Settings/ViewControllers/AddRPCServerViewController.swift

```diff
@@ -0,0 +1,276 @@
+        ] + roundedBackground.createConstraintsWithContainer(view: view))
+
+        hidesBottomBarWhenPushed = true
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        buttonsBar.configure()
+        networkNameTextField.configureOnce()
+        rpcUrlTextField.configureOnce()
+        chainIDTextField.configureOnce()
+        symbolTextField.configureOnce()
+        blockExplorerURLTextField.configureOnce()
+        isTestNetworkView.configure(viewModel: viewModel.enableServersHeaderViewModel)
+
+        buttonsBar.buttons[0].addTarget(self, action: #selector(saveCustomRPC), for: .touchUpInside)
+
+        configure(viewModel: viewModel)
+
+        let tap = UITapGestureRecognizer(target: self, action: #selector(tapSelected))
+        roundedBackground.addGestureRecognizer(tap)
+    }
+
+    @objc private func tapSelected(_ sender: UITapGestureRecognizer) {
+        view.endEditing(true)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        keyboardChecker.viewWillAppear()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        keyboardChecker.viewWillDisappear()
+    }
+
+    required init?(coder: NSCoder) {
+        return nil
+    }
+
+    func configure(viewModel: AddrpcServerViewModel) {
+        navigationItem.title = viewModel.title
+
+        networkNameTextField.label.text = viewModel.networkNameTitle
+        rpcUrlTextField.label.text = viewModel.rpcUrlTitle
+        chainIDTextField.label.text = viewModel.chainIDTitle
+        symbolTextField.label.text = viewModel.symbolTitle
+        blockExplorerURLTextField.label.text = viewModel.blockExplorerURLTitle
+
+        buttonsBar.buttons[0].setTitle(viewModel.saverpcServerTitle, for: .normal)
+    }
+
+    private func validateInputs() -> Bool {
+        var isValid: Bool = true
+
+        if networkNameTextField.value.isEmpty {
+            isValid = false
+            networkNameTextField.status = .error(R.string.localizable.addrpcServerNetworkNameError())
+        } else {
+            networkNameTextField.status = .none
+        }
+
+        if URL(string: rpcUrlTextField.value) == nil {
+            isValid = false
+            rpcUrlTextField.status = .error(R.string.localizable.addrpcServerRpcUrlError())
+        } else {
+            rpcUrlTextField.status = .none
+        }
+
+        if Int(chainIDTextField.value) == nil {
+            isValid = false
+            chainIDTextField.status = .error(R.string.localizable.addrpcServerChainIDError())
+        } else {
+            chainIDTextField.status = .none
+        }
+
+        if symbolTextField.value.isEmpty {
+            isValid = false
+            symbolTextField.status = .error(R.string.localizable.addrpcServerSymbolError())
+        } else {
+            symbolTextField.status = .none
+        }
+
+        if URL(string: blockExplorerURLTextField.value) == nil {
```

Comment:
Yeah, let's let this be mandatory for now, although it can be optional strictly speaking and see if users feedback.